### PR TITLE
refactor(is-focusable): account for negative tabindex

### PR DESCRIPTION
**Pull request by dbowling**
- add option arg that enables looking for negative tabindex (defaults to false)

references: #3500

### CODE REVIEW

File: lib/commons/dom/is-focusable.js [hunk 1]
```diff
@@ -9,22 +9,33 @@ import { getNodeFromTree } from '../../core/utils';
  * @memberof axe.commons.dom
  * @instance
  * @param {HTMLElement} el The HTMLElement
+ * @param {object} options Optional. If set to {programatically: false}, the function will return true
  * @return {Boolean} The element's focusability status
  */
-export default function isFocusable(el) {
+export default function isFocusable(el, options = { programmatically: false }) {
```
Comment (straker) [inline lib/commons/dom/is-focusable.js:15]: No need to set the default

```suggestion
export default function isFocusable(el, options = {}) {
```

File: lib/commons/dom/is-focusable.js [hunk 2]
```diff
@@ -9,22 +9,33 @@ import { getNodeFromTree } from '../../core/utils';
...
   if (vNode.props.nodeType !== 1) {
     return false;
   }
 
+  var tabindex = vNode.attr('tabindex');
+
+  if (
+    tabindex &&
+    options.programmatically === true &&
+    parseInt(tabindex, 10) === -1
```
Comment (straker) [inline lib/commons/dom/is-focusable.js:27]: `-1` is not the only valid negative tabindex value

```suggestion
    parseInt(tabindex, 10) < 0
```

**Review by straker — DISMISSED** [review]

File: lib/commons/dom/is-focusable.js (hunk 2 above)
Comment (straker) [reply lib/commons/dom/is-focusable.js:27]: You'll also want to modify one of the tests for another negative value other than `-1`

File: lib/commons/dom/is-focusable.js [hunk 3]
```diff
@@ -9,22 +9,33 @@ import { getNodeFromTree } from '../../core/utils';
...
   }
 
+  var tabindex = vNode.attr('tabindex');
+
+  if (
+    tabindex &&
+    options.programmatically === true &&
+    parseInt(tabindex, 10) < 0
+  ) {
+    return false;
```
Comment (WilcoFiers) [inline lib/commons/dom/is-focusable.js:29]: Elements with a negative tabindex are programmatically focucsable.

```suggestionI
    return true;
```

**Review by WilcoFiers — CHANGES_REQUESTED** [review]
1. This is not a feature, it's a refactor
2. Please use the option, otherwise you've just added dead code.

File: test/commons/dom/is-focusable.js [hunk 4]
```diff
@@ -184,89 +184,221 @@ describe('is-focusable', function() {
...
+    it('should return false for a div with no tabindex', function () {
       fixture.innerHTML = '<div id="target"></div>';
       var el = document.getElementById('target');
       flatTreeSetup(fixture);
 
       assert.isFalse(axe.commons.dom.isFocusable(el));
     });
+    describe('options', function () {
+      describe('programatically', function () {
+        describe('true (default)', function () {
```
Comment (WilcoFiers) [inline test/commons/dom/is-focusable.js:196]: This is kind of a personal preference, but when it comes to boolean options I prefer defaults to be falsey. It's not intuitive (to me) for undefined and true to behave the same, but for false to be different. Much of axe-core works that way, so it's more consistent doing this too.

I propose using `ifProgrammatic` with a falsey default.

**Review by WilcoFiers — DISMISSED** [review]

**Comment by dbowling** [comment]
Just documenting a discussion for this PR:

We're going to create a new function and leave isFocusable unchanged. This should drastically simplify future implementations. Also, implementations will go into a new issue as needed to avoid scope creep.

I'll force push the new approach, so I'm resolving all requests for changes now as outdated.

**Comment by straker** [comment]
We seem to have somehow confused the GitHub UI. This pr was not merged into develop, does [not appear in develops commit history](https://github.com/dequelabs/axe-core/commits/develop), and GitHub even says 0 commits were merged. The commit that says was merged (49ccb00) is [an entirely different pr](https://github.com/dequelabs/axe-core/commit/49ccb007c63cc3f1dce7b82cb7630856fb17605c). Not sure what happened.

Screenshot from GitHub showing: 
dbowling merged 0 commits into develop from 3500-isFocusable-negative-tabindex
